api/src/main/java/com/datatorrent/api/Operator.java

```diff
@@ -172,6 +172,51 @@ public String toString()
 
   }
 
+  public interface ProxyPort<T> extends Port
+  {
+    void set(T port);
+
+    T get();
+  }
+
+  public final class ProxyInputPort<T> extends DefaultInputPort<T> implements ProxyPort<InputPort<T>>
```

Comment:
This is because DefaultInputPort also implements InputPort interface. If we implement it for ProxyInputPort, we'll anyway have to again implement the methods that DefaultInputPort is anyway implementing. Additionally, this is just a placeholder until the InputPort object inside ProxyInputPort is set. We have also made it final so that no one can extend it.